nuxeo-coldstorage/src/main/java/org/nuxeo/coldstorage/service/ColdStorageServiceImpl.java

```diff
@@ -275,28 +349,49 @@ public static DocumentModel restoreContentFromColdStorage(CoreSession session, D
         return documentModel;
     }
 
-    /**
-     * Gets the number of the days where the document's blob is available, once it's retrieved from cold storage.
-     *
-     * @return the number of days of availability if property
-     *         {@value COLD_STORAGE_NUMBER_OF_DAYS_OF_AVAILABILITY_PROPERTY_NAME} is configured, {@code 1} otherwise.
-     * @since 10.10
-     */
-    public static int getNumberOfDaysOfAvailability() {
-        String value = Framework.getProperties()
-                                .getProperty(COLD_STORAGE_NUMBER_OF_DAYS_OF_AVAILABILITY_PROPERTY_NAME, "1");
-        return Integer.parseInt(value);
+    @Override
+    public void restoreMainContent(DocumentModel documentModel) {
+        CoreSession session = documentModel.getCoreSession();
+        DocumentEventContext ctx = new DocumentEventContext(session, session.getPrincipal(), documentModel);
+        EventService eventService = Framework.getService(EventService.class);
+        eventService.fireEvent(ctx.newEvent(COLD_STORAGE_CONTENT_TO_RESTORE_EVENT_NAME));
+    }
+
+    @Override
+    public void moveDuplicatedBlobToColdStorage(CoreSession session, DocumentModel documentModel) {
+        String blobDigest = ((Blob) documentModel.getPropertyValue(COLD_STORAGE_CONTENT_PROPERTY)).getDigest();
+        String documentId = documentModel.getId();
+        String query = String.format(
+                "SELECT * FROM Document WHERE (ecm:isVersion = 0 AND file:content/digest = '%s') "
+                        + "OR (ecm:isVersion = 1 AND ecm:versionVersionableId = '%s' AND file:content/digest = '%s')",
+                blobDigest, documentModel.getId(), blobDigest);
+
+        BulkService bulkService = Framework.getService(BulkService.class);
+        BulkCommand.Builder builder = new BulkCommand. //
+                Builder(DeduplicationColdStorageContentActions.ACTION_NAME, query) //
+                                                                                  .user(SecurityConstants.SYSTEM_USERNAME)
+                                                                                  .repository(
+                                                                                          session.getRepositoryName());
+
+        BulkCommand bulkCommand = builder.build();
+        String commandId;
+        try {
+            commandId = bulkService.submit(bulkCommand);
+        } catch (IllegalArgumentException e) {
+            throw new NuxeoException(e.getMessage(), e, SC_BAD_REQUEST);
```

Comment:
As the BulkCommand is built here in the code, you shouldn't have this exception, do you?

Reply:
> As the BulkCommand is built here in the code, you shouldn't have this exception, do you?

I will remove it.